lib/libspl/os/linux/getexecname.c

```diff
@@ -19,41 +19,15 @@
  *
  * CDDL HEADER END
  */
-/*
- * Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
- * Use is subject to license terms.
- */
-
 
 #include <limits.h>
-#include <pthread.h>
-#include <stdlib.h>
-#include <string.h>
+#include <stdint.h>
 #include <unistd.h>
 
-const char *
-getexecname(void)
-{
-	static char execname[PATH_MAX + 1] = "";
-	static pthread_mutex_t mtx = PTHREAD_MUTEX_INITIALIZER;
-	char *ptr = NULL;
-	ssize_t rc;
+extern ssize_t getexecname_impl(char *execname);
```

Comment:
I'd suggest moving all of the extern declarations to `lib/libspl/include/stdlib.h` so they can be property type checked.

Reply:
Hm, good point; I've opted for a new `libspl_impl.h` header instead so as to not pollute the global namespace with an unresolvable internal name